carla_ego_vehicle/src/carla_ego_vehicle/carla_ego_vehicle.py

```diff
@@ -79,12 +113,20 @@ def __init__(self):
             pose.orientation.w = quat[3]
             self.actor_spawnpoint = pose
 
-        self.initialpose_subscriber = rospy.Subscriber(
-            "/carla/{}/initialpose".format(self.role_name),
-            PoseWithCovarianceStamped,
-            self.on_initialpose)
-        rospy.loginfo('listening to server %s:%s', self.host, self.port)
-        rospy.loginfo('using vehicle filter: %s', self.actor_filter)
+        self.initialpose_subscriber = self.create_subscriber(PoseWithCovarianceStamped,
```

Comment:
limitation we use only the default network policy to be queue_size=10. We can leave it for now but this will need to be more generic, and define a QoS for ros2 case

Reply:
`CompatibleNode.create_subsciber()` accepts arbitrary QoS-profile as an argument. If none is given the default is used.